src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/DedupJobSchedulerDAO.java

```diff
@@ -0,0 +1,141 @@
+        "          reservedIndex = reservedIndex + 2\n" +
+        "          redis.call('ZREM', $readyQueue$, jobs[i])\n" +
+        "          redis.call('ZADD', $runningQueue$, $ttl$, jobs[i])\n" +
+        "          nextLimit = nextLimit - 1\n" +
+        "      end\n" +
+        "   end\n" +
+        "   nextOffset = nextOffset + nextLimit\n" +
+        "end\n" +
+        "return reserved"
+    )
+    List<TimeJobInfo> reserveJobs(
+            @BindKey("readyQueue") String readyQueue,
+            @BindKey("runningQueue") String runningQueue,
+            @BindKey("pausedTube") String pausedTube,
+            @BindArg("limit") int limit,
+            @BindArg("now") long now,
+            @BindArg("ttl") long ttl);
+
+    /**
+     * Jobs in the ready queue are scored/sorted by when they should run. A ready job
+     * is considered expired if its score/scheduled run time is less than equal to the
+     * expirationTimeInMillis. As a result expirationTimeInMillis should be set to
+     * now - (the threshold past which a job sitting waiting to be reserved is considered expired).
+     *
+     * Jobs in the running queue are sorted by their time to live (ttl). A running job
+     * is considered expired if the current time is past the ttl. So expirationTimeInMillis
+     * should be set to now.
+     */
+    @Mapper(TimeJobInfoListMapper.class)
+    @Query(
+        "local expiredJobs = redis.call('ZRANGEBYSCORE', $queue$, 0, $expirationTimeInMillis$, 'WITHSCORES')\n" +
+        "redis.call('ZREMRANGEBYSCORE', $queue$, 0, $expirationTimeInMillis$)\n" +
+        "return expiredJobs"
+    )
+    List<TimeJobInfo> removeExpiredJobs(@BindKey("queue") String queue,
+                                        @BindArg("expirationTimeInMillis") Long expirationTimeInMillis);
+
+    /**
+     * Deleting a job deletes it from the scheduler in its entirety.
+     */
+    @Query(
+        "local deletedFromReadyQueue = redis.call('ZREM', $readyQueue$, $job$)\n" +
+        "local deletedFromRunningQueue = redis.call('ZREM', $runningQueue$, $job$)\n" +
+        "if deletedFromReadyQueue == 0 and deletedFromRunningQueue == 0 then\n" +
+        "   return 0\n" +
+        "else\n" +
+        "   return 1\n" +
+        "end"
+    )
+    int deleteJob(
+            @BindKey("readyQueue") String readyQueue,
+            @BindKey("runningQueue") String runningQueue,
+            @BindArg("job") String job);
+
+    /**
+     * Ack indicates we are done with a reserved job that is in the running queue.
+     * Another instance of that job might already be queued up in the ready queue waiting
+     * to be reserved.
+     */
+    @Query(
+        "return redis.call('ZREM', $runningQueue$, $job$)"
+    )
+    int ackJob(
+            @BindKey("runningQueue") String runningQueue,
+            @BindArg("job") String job);
+
+    @Query(
+        "local inQueue = redis.call('ZSCORE', $queue$, $job$)\n" +
+        "if inQueue then\n" +
```

Comment:
this is the score. it will return a null if there is no score for the job. in the delete we are actually doing a zrem which returns the number removed. so different type of return values